src/types.rs

```diff
@@ -13,41 +13,56 @@ pub struct Response {
     pub remaining_points: u64,
 }
 
-#[derive(Clone)]
-pub struct Interface {
+pub trait InterfaceClone {
```

Comment:
You define that trait so interface can kinda implement clone ?

Reply:
Yes, so we can implement `Clone` for `Box< dyn Interface>`, so that it can be a member of all those structures that derive `Clone`.